### PR TITLE
GitHub hosted charts

### PR DESCRIPTION
**Pull request by chrisbsmith**
This PR doesn't bring any major change to the actual charts themselves, instead, it moves where we archive and host the charts. Rather than using Nexus to host the charts, this PR introduces the [chart release action](https://github.com/helm/chart-releaser-action) that runs and hosts the charts in GitHub. To do this, I had to create a new, empty branched called `gh-pages`. The chart-releaser action packages the charts, stores them at releases in GitHub, and creates an `index.yaml` that points the released charts.

The charts are now available at `https://greymatter-io.github.io/helm-charts`

ℹ️ You can actually do this already since I tested it.  Just run `helm repo add greymatter https://greymatter-io.github.io/helm-charts`

There's a second action that I added in this PR to sync the README from the default branch to the `gh-pages` branch so that there is something there to read when users go to that branch.

I updated the README to point to the GitHub hosted charts instead of running locally.    

TODO: 
- once we're satisfied with how GitHub hosts charts, we can remove the CircleCI jobs that package and push the charts there.
- update all documentation how to pull the charts (I wanted to get this workflow out first and we can bring the documentation up to speed later)

@justincely FYI, in case we need to update official Grey Matter documentation

### CODE REVIEW

**Comment by moosecanswim** [comment]
LGTM.  I added the  repo and pulled down the edge chart and rand a template all as expected.

File: README.md [hunk 1]
```diff
@@ -1,7 +1,57 @@
...
+
+```console
+make credentials
+```
+
+### Installing
+
+The following set of commands will install Grey Matter using the GitHub hosted Helm Charts
+
+```console
```
Comment (moosecanswim) [inline README.md:36]: These can also be accomplished using the make targets

**Comment by tilleryd** [comment]
This sounds amazing. Did you consider [git LFS](https://git-lfs.github.com/) as well?

**Comment by chrisbsmith** [comment]
@tilleryd, no I didn't consider git LFS.  The helm chart-releaser-action just supports the releases right now.  This is pretty much an out-of-the-box solution.

File: README.md (hunk 1 above)
Comment (chrisbsmith) [reply README.md:36]: @moosecanswim I think the make targets are a bit different and they specifically call the local charts.

Comment (moosecanswim) [reply README.md:36]: youre right

**Review by moosecanswim — APPROVED** [review]
LGTM, works as expected and you addressed my comments

File: .github/workflows/sync-readme.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# gh-pages branch
+on:
+  push:
+    branches:
+      - 'release-*'
+    paths:
+      - 'README.md'
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (tilleryd) [inline .github/workflows/sync-readme.yaml:15]: Why not run on `alpine`?

File: README.md [hunk 3]
```diff
@@ -1,7 +1,57 @@
...
+helm repo add greymatter https://greymatter-io.github.io/helm-charts
+```
+
+You can then run `helm search repo greymatter` to see the charts.
+
+## Quickstart with Grey Matter
+
+The steps below provide a quickstart exmple to deploy Grey Matter.
```
Comment (tilleryd) [inline README.md:22]:
```suggestion
The steps below provide a quickstart example to deploy Grey Matter.
```

File: README.md [hunk 4]
```diff
@@ -1,7 +1,57 @@
...
+
+### Create Credentials
+
+A Grey Matter LDAP account is still required to pull the images from our Nexus server. Run the following command and provide answers when prompted
```
Comment (tilleryd) [inline README.md:26]:
```suggestion
A Grey Matter LDAP account is still required to pull the images from our Nexus server. Run the following command and provide answers when prompted.
```

File: README.md [hunk 5]
```diff
@@ -1,7 +1,57 @@
...
+
+```console
+make credentials
+```
+
+### Installing
+
+The following set of commands will install Grey Matter using the GitHub hosted Helm Charts
```
Comment (tilleryd) [inline README.md:34]:
```suggestion
The following set of commands will install Grey Matter using the GitHub hosted Helm Charts.
```

File: .github/workflows/sync-readme.yaml (hunk 2 above)
Comment (chrisbsmith) [reply .github/workflows/sync-readme.yaml:15]: This is what the example I used followed.  Not sure it really makes a difference for us though

File: README.md [hunk 6]
```diff
@@ -1,7 +1,57 @@
...
+
+```console
+$ helm install secrets greymatter/secrets -f credentials.yaml -f global.yaml
+$ helm install spire greymatter/spire --set=global.environment=kubernetes -f global.yaml
+$ helm install fabric greymatter/fabric --set=global.environment=kubernetes -f global.yaml
+$ helm install edge greymatter/edge -f global.yaml
+$ helm install sense greymatter/sense -f global.yaml --set=global.waiter.service_account.create=false
+$ kubectl patch svc edge -p '{"spec": {"type": "LoadBalancer"}}'
```
Comment (tilleryd) [inline README.md:42]: Zoe made the edge service type configurable, so you can now do `helm install edge greymatter/edge --set=edge.ingress.type=LoadBalancer -f global.yaml` in the third to last command.  See https://github.com/greymatter-io/helm-charts/pull/769

File: README.md [hunk 7]
```diff
@@ -1,7 +1,57 @@
...
+```
+
+### Viewing the Grey Matter Dashboard
```
Comment (tilleryd) [inline README.md:45]:
```suggestion
### Viewing the Grey Matter Application
```

This change follows the new name for the dashboard. See https://github.com/DecipherNow/gm-gitbook-sync/pull/284

File: README.md (hunk 6 above)
Comment (chrisbsmith) [reply README.md:42]: Nice, thanks. Updating coming

Comment (tilleryd) [reply README.md:42]: I should've just made a suggestion... sorry

Comment (chrisbsmith) [reply README.md]: No worries, update pushed

**Review by tilleryd — APPROVED** [review]
LGTM

**Comment by chrisbsmith** [comment]
Merging this with a failed test as these changes don't impact actual helm chart code.  Will investigate the failed test further.

**Pull request merged**